packages/gatsby-theme-docs/src/hooks/use-layout-state.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+
+const useLayoutState = () => {
```

Comment:
I moved all this state into a hook, to make is easier to reuse across the different layout components.